lib/dsl_prompt_transpiler.py

```diff
@@ -97,13 +86,23 @@ def negative_weight_expr(self, args):
         return ast.WeightedExpression(*args, positive=False)
 
     @v_args(inline=True)
-    def interpolation_expr(self, exprs, steps, function_name=None):
-        function_name = str(function_name) if function_name is not None else None
+    def interpolation_expr(self, subexprs):
+        if str(subexprs[-1]) in {'linear', 'catmull', 'bezier'}:
+            function_name = str(subexprs[-1])
+            subexprs = subexprs[:-1]
+        else:
+            function_name = None
+
+        steps = [None if not step
+                 else ast.SubstitutionExpression(step[1:]) if step.startswith('$')
+                 else ast.LiftExpression(step)
+                 for step in str(subexprs[-1]).split(',')]
```

Comment:
to make the interpolation parsing work I had to dumb it out here. We are going back and forth `str` -> `*Expression` -> `str` -> `*Expression`, which is a bit inefficient. This isn't an immediate problem, but I had to implement a bunch of `__str__` methods for this to work, which is code we don't want to maintain.

Reply:
This isn't too bad as the syntax won't change in the near future. This will be easier to address when implementing #16